ebean-datasource/src/main/java/io/ebean/datasource/pool/PooledConnection.java

```diff
@@ -700,6 +700,10 @@ public void setSchema(String schema) throws SQLException {
     if (status == STATUS_IDLE) {
       throw new SQLException(IDLE_CONNECTION_ACCESSED_ERROR + "setSchema()");
     }
+    if (originalSchema == null) {
+      // lazily initialise the originalSchema
+      originalSchema = getSchema();
```

Comment:
the comment above makes me think critically again at this point

We would need some additional magic here, that we can detect the case, when `getSchema()` returns null the first time.

When we call `setSchema("S1")`, we will get `null` form `getSchema()`
When we then call `setSchema("S2")` we will set `S1` as originalSchema, which is wrong.

So we would either need additional booleans (`originalSchemaRead`) here or some magic string object here
`private static final String UNSET = new String("__UNSET__")` and check `if (originalSchema == UNSET)` (do not use .equals here)
But I think booleans are more understandable and bullet proof here.

Reply:
Added initialisedSchema and initialisedCatalog boolean flags to handle the case for null returned from getSchema() / getCatalog().

Should be good now.